File: minji/week1/10845.cpp
/*
	[BOJ] 10845 ť
*/
#include <iostream>
#include <queue>
#define endl "\n"

using namespace std;
queue<int> q;
int N;
int main() {
	ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
	cin >> N;
	while (N) {
		string command;
		cin >> command;
		//while (getchar() != '\n');
		int num = 0;
		if (command == "push") {
			cin >> num;
			q.push(num);

		}
		else if (command == "pop") {
			if (q.size() == 0) {
				cout << "-1" << endl;
			}
			else {
				int num = q.front();
				q.pop();
				cout << num << endl;
			}
		}
		else if (command == "size") {
			cout << q.size() << endl;
		}
		else if (command == "empty") {
			if (q.size() > 0) {
				cout << q.empty() << endl;
			}
			else {
				cout << "1" << endl;
			}
		}
		else if (command == "front") {
			if (q.size() > 0) {
				cout << q.front() << endl;
			}
			else {
				cout << "-1" << endl;
			}
		}
		else if (command == "back") {
			if (q.size() > 0) {
				cout << q.back() << endl;
			}
			else {
				cout << "-1" << endl;
			}
		}
		--N;
	}

	return 0;
}

File: minji/week1/10845.py
'''
    [BOJ] 10845 큐 
'''
import sys
N = int(sys.stdin.readline().rstrip())
q = []

for _ in range(N):
    cmd = sys.stdin.readline().rstrip().split()
    if cmd[0] == "push":
        q.append(cmd[1])
    elif cmd[0] == "pop":
        if (not q):
            print(-1)
        else:
            print(q.pop(0))
    elif cmd[0] == "size":
        print(len(q))
    elif cmd[0] == "empty":
        if q:
            print(0)
        else:
            print(1)
    elif cmd[0] == "front":
        if q:
            print(q[0])
        else:
            print(-1)
    elif cmd[0] == "back":
        if q:
            print(q[-1])
        else:
            print(-1)


